backend/lib/edgehog/devices/device/manual_actions/send_create_container.ex

```diff
@@ -0,0 +1,64 @@
+#
+# This file is part of Edgehog.
+#
+# Copyright 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+defmodule Edgehog.Devices.Device.ManualActions.SendCreateContainer do
+  @moduledoc false
+  use Ash.Resource.ManualUpdate
+
+  @send_create_container_request_behaviour Application.compile_env(
+                                             :edgehog,
+                                             :astarte_create_container_request_module,
+                                             Edgehog.Astarte.Device.CreateContainerRequest
+                                           )
+
+  @impl Ash.Resource.ManualUpdate
+  def update(changeset, _opts, _context) do
+    device = changeset.data
+
+    with {:ok, container} <- Ash.load(changeset.container, :env_encoding) do
+      env_encoding = container.env_encoding
+      restart_policy = to_correct_string(container.restart_policy)
+
+      data = %{
```

Comment:
```suggestion
      data = %RequestData{
```

I think dyalizer wouldn't like this as it is. Let's give it "proper" typing

Reply:
The `send_datastream` function accepts a map as input, not a `RequestData` instance. For simplicity I intially just wrote a map but proer typing is better I agree

Reply:
It works, but it breaks the contract: [`9f02250` (#646)](https://github.com/edgehog-device-manager/edgehog/pull/646/commits/9f02250a6df6a69c2a9ac62d7624ba485dfc78e8#diff-b705cafe3b60fe47455496a41accaa2ac6316f9c2ac96d2ddd0dd70fb6b94db7R29)